### PR TITLE
[SYCL][Doc] Update design doc for new offloading model

### PR DESCRIPTION
**Pull request by asudarsa**
Adds information about the following options:
1. sycl-post-link-options
2. llvm-spirv-options

Thanks
Sincerely

### CODE REVIEW

File: sycl/doc/design/OffloadDesign.md [hunk 1]
```diff
@@ -208,9 +208,20 @@ The `sycl-post-link` tool is used after the device link is performed,
...
+| `--llvm-spirv-options=<arg>`         | Options that will control llvm-spirv step |
+
+*Table: Options to pass sycl-post-link and llvm-spirv options to the clang-linker-wrapper*
+
+Options that will be used by clang-linker-wrapper when invoking `sycl-post-link`
+tool are provided by the driver via the `--sycl-post-link-options=<arg>` option.
+Options that will be used by clang-linker-wrapper when invoking `llvm-spirv`
+tool are provided by the driver via the `--llvm-spirv-options=<arg>` option.
+
+SPIR-V binary is thn sent to the Ahead of Time compilation step to produce the
```
Comment (bader) [inline sycl/doc/design/OffloadDesign.md:223]:
```suggestion
SPIR-V binary is then sent to the Ahead of Time compilation step to produce the
```

**Comment by bader** [comment]
This is not related to this patch, but "General Offload Flow" diagram has "unbundle" step, which is not covered in the text. I think it's an artifact of updating current offload flow. Am I right?

**Comment by asudarsa** [comment]
> This is not related to this patch, but "General Offload Flow" diagram has "unbundle" step, which is not covered in the text. I think it's an artifact of updating current offload flow. Am I right?

Hi @bader 

Thanks for the question. Yes. The getDeviceInput routine in clang-linker-wrapper will be able to fetch all the device offload images. There is no need to unbundle. However, I do have logic to unbundle in current flow to handle any 'old' bundles.

Thanks

**Comment by asudarsa** [comment]
Hi @RaviNarayanaswamy and @mdtoguchi 

Can you please take a look?

Thanks

File: sycl/doc/design/OffloadDesign.md [hunk 2]
```diff
@@ -208,9 +208,20 @@ The `sycl-post-link` tool is used after the device link is performed,
...
-step to produce the final device binary or is wrapped before being linked into
-the final executable.
+|            Option                    |   Expected Behavior   |
+|--------------------------------------|-----------------------|
+| `--sycl-post-link-options=<arg>`     | Options that will control sycl-post-link step |
+| `--llvm-spirv-options=<arg>`         | Options that will control llvm-spirv step |
+
+*Table: Options to pass sycl-post-link and llvm-spirv options to the clang-linker-wrapper*
+
+Options that will be used by clang-linker-wrapper when invoking `sycl-post-link`
```
Comment (mdtoguchi) [inline sycl/doc/design/OffloadDesign.md:218]:
```suggestion
Options that will be used by clang-linker-wrapper when invoking the `sycl-post-link`
```

File: sycl/doc/design/OffloadDesign.md [hunk 3]
```diff
@@ -208,9 +208,20 @@ The `sycl-post-link` tool is used after the device link is performed,
...
+tool are provided by the driver via the `--sycl-post-link-options=<arg>` option.
+Options that will be used by clang-linker-wrapper when invoking `llvm-spirv`
+tool are provided by the driver via the `--llvm-spirv-options=<arg>` option.
+
+SPIR-V binary is then sent to the Ahead of Time compilation step to produce the
+final device binary or is wrapped before being linked into the final executable.
```
Comment (mdtoguchi) [inline sycl/doc/design/OffloadDesign.md:224]: This isn't clear - we always wrap before being linked into the final executable.  The AOT step is the optional one based on the expected target.

File: sycl/doc/design/OffloadDesign.md (hunk 2 above)
Comment (asudarsa) [reply sycl/doc/design/OffloadDesign.md]: Done. Thanks

File: sycl/doc/design/OffloadDesign.md [hunk 4]
```diff
@@ -208,9 +208,20 @@ The `sycl-post-link` tool is used after the device link is performed,
...
+|            Option                    |   Expected Behavior   |
+|--------------------------------------|-----------------------|
+| `--sycl-post-link-options=<arg>`     | Options that will control sycl-post-link step |
+| `--llvm-spirv-options=<arg>`         | Options that will control llvm-spirv step |
+
+*Table: Options to pass sycl-post-link and llvm-spirv options to the clang-linker-wrapper*
+
+Options that will be used by clang-linker-wrapper when invoking `sycl-post-link`
+tool are provided by the driver via the `--sycl-post-link-options=<arg>` option.
+Options that will be used by clang-linker-wrapper when invoking `llvm-spirv`
```
[the remark this answers is not in the document]
Comment (asudarsa) [reply sycl/doc/design/OffloadDesign.md]: Done

File: sycl/doc/design/OffloadDesign.md (hunk 3 above)
Comment (asudarsa) [reply sycl/doc/design/OffloadDesign.md]: We actually do not need to mention this. Removing the sentence.

Thanks

**Review by mdtoguchi — APPROVED** [review]

File: sycl/doc/design/OffloadDesign.md [hunk 5]
```diff
@@ -208,9 +208,17 @@ The `sycl-post-link` tool is used after the device link is performed,
 applying any changes such as optimizations and code splitting before passing
 off to the `llvm-spirv` tool, which translates the LLVM-IR to SPIR-V.
 
-At this point, the SPIR-V binary is sent to the Ahead of Time compilation
-step to produce the final device binary or is wrapped before being linked into
-the final executable.
+|            Option                    |   Expected Behavior   |
```
Comment (RaviNarayanaswamy) [inline sycl/doc/design/OffloadDesign.md:211]: Are there any options that are need to be passed to ocloc for AOT.

Comment (asudarsa) [reply sycl/doc/design/OffloadDesign.md:211]: We have already added them in the text below this change.

Thanks

**Review by RaviNarayanaswamy — APPROVED** [review]

**Comment by asudarsa** [comment]
Hi @bader 

Can you please take a look and merge if ready?

Thanks

**Pull request merged**